File: ruby/spec/miscreant/internals/block_spec.rb
# encoding: binary
# frozen_string_literal: true

RSpec.describe Miscreant::Internals::Block do
  describe "dbl" do
    let(:test_vectors) { DblExample.load_file }

    it "passes all test vectors" do
      test_vectors.each do |ex|
        block = described_class.new(ex.input)
        block.dbl
        expect(block.data).to eq(ex.output)
      end
    end
  end
end

File: js/test/block.spec.ts
import { suite, test } from "mocha-typescript";
import { expect } from "chai";
import Block from "../src/internals/block";

import { DblExample } from "./support/test_vectors";

@suite class DblSpec {
  static vectors: DblExample[];

  static async before() {
    this.vectors = await DblExample.loadAll();
  }

  @test async "passes the AES-CMAC test vectors"() {
    let block = new Block();

    for (let v of DblSpec.vectors) {
      block.data.set(v.input);
      block.dbl();
      expect(block.data).to.eql(v.output);
    }
  }
}
